src/test/groovy/org/prebid/server/functional/tests/pg/TokenSpec.groovy

```diff
@@ -19,6 +20,51 @@ class TokenSpec extends BasePgSpec {
         pgPbsService.sendForceDealsUpdateRequest(ForceDealsUpdateRequest.invalidateLineItemsRequest)
     }
 
+    def "PBS should prioritize line item when pg ignore pacing and #reason"() {
+        given: "Bid request"
+        def bidRequest = BidRequest.defaultBidRequest
+
+        and: "Planner Mock with additional lineItem"
+        def plansResponse = PlansResponse.getDefaultPlansResponse(bidRequest.site.publisher.id).tap {
+            lineItems.add(updateLineItem(bidRequest.site.publisher.id))
+        }
+        generalPlanner.initPlansResponse(plansResponse)
+
+        and: "Bid response"
+        def bidResponse = BidResponse.getDefaultPgBidResponse(bidRequest, plansResponse)
+        bidder.setResponse(bidRequest.id, bidResponse)
+
+        and: "Line items are fetched by PBS"
+        updateLineItemsAndWait()
+
+        and: "Pg ignore pacing header"
+        def pgIgnorePacingHeader = ["${HttpUtil.PG_IGNORE_PACING_HEADER}": "1"]
+
+        when: "Auction is requested"
+        def auctionResponse = pgPbsService.sendAuctionRequest(bidRequest, pgIgnorePacingHeader)
+
+        then: "PBS should process PG deals"
+        def pgMetrics = auctionResponse.ext?.debug?.pgmetrics
+        for(i in 0..< pgMetrics.readyToServe.size()) {
```

Comment:
Please make sure that your size is not 0. And you can then use better construct instead of for:
```
pgMetrics.readyToServe.eachWithIndex { it, index ->
    it == plansResponse.lineItems[index].lineItemId
```